### PR TITLE
Add length to constant config

### PR DESCRIPTION
**Pull request by LucaTomasko**
Issue #196

### CODE REVIEW

File: tests/capture/input_generators/test_attack_point_iterator.py [hunk 1]
```diff
@@ -293,7 +358,8 @@ def test_attack_point_iterator_repeat_infinite_minus_two():
         "configuration": {
             "operation": "constants",
             "name": "key",
-            "values": [1, 2]
+            "length": 2,
+            "values": [[1, 2]]
```
Comment (kralka) [inline tests/capture/input_generators/test_attack_point_iterator.py:362]: This changes the unit-test. Before we were iterating
1
2
1
2
1
2
...

Now we are iterating always the same value.
[1,2]
[1,2]
[1,2]
...

Luckily there is another test which is covering that.

**Review by kralka — COMMENTED** [review]
Thank you, very nice.

File: scaaml/capture/input_generators/attack_point_iterator.py [hunk 2]
```diff
@@ -81,12 +77,30 @@ def _build_attack_points_iterator(configuration: Dict) -> AttackPointIterator:
...
                  values: List[List[int]]) -> None:
-        """Initialize the constants to iterate."""
+        """Initialize the constants to iterate.
+        
+            Args:
+                operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                name (str): The name represents the key name of the value.
```
Comment (kralka) [inline scaaml/capture/input_generators/attack_point_iterator.py:89]:
```suggestion

                name (str): The name represents the key name of the value.
```

File: scaaml/capture/input_generators/attack_point_iterator.py [hunk 3]
```diff
@@ -81,12 +77,30 @@ def _build_attack_points_iterator(configuration: Dict) -> AttackPointIterator:
...
+                length (int): The prescribed length for each list in values.
```
Comment (kralka) [inline scaaml/capture/input_generators/attack_point_iterator.py:90]:
```suggestion

                length (int): The prescribed length for each list in values.
```

File: scaaml/capture/input_generators/attack_point_iterator.py [hunk 4]
```diff
@@ -81,12 +77,30 @@ def _build_attack_points_iterator(configuration: Dict) -> AttackPointIterator:
...
-    def __init__(self, operation: str, name: str,
+    def __init__(self, operation: str, name: str, length: int,
                  values: List[List[int]]) -> None:
-        """Initialize the constants to iterate."""
+        """Initialize the constants to iterate.
+        
+            Args:
+                operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
```
Comment (kralka) [inline scaaml/capture/input_generators/attack_point_iterator.py:87]:
```suggestion
                arguments should be present. This is only used once to
```

Comment (kralka) [reply scaaml/capture/input_generators/attack_point_iterator.py:87]: This class has no config file. It does not touch a config file. Not even the user-facing build_attack_points_iterator is concerned with files.

File: scaaml/capture/input_generators/attack_point_iterator.py (hunk 2 above)
Comment (kralka) [reply scaaml/capture/input_generators/attack_point_iterator.py:89]: Paragraphs make the designdoc easier to read. Or indentation works too.

File: scaaml/capture/input_generators/attack_point_iterator.py [hunk 5]
```diff
@@ -81,12 +77,30 @@ def _build_attack_points_iterator(configuration: Dict) -> AttackPointIterator:
...
+                double check if the operation is the correct one.
+                name (str): The name represents the key name of the value.
+                length (int): The prescribed length for each list in values.
+                If one of the lists isn't the same length as this variable
+                it will raise an ListNotPrescribedLengthException.
+                values (List[List[int]]): List of lists of ints that gets
```
Comment (kralka) [inline scaaml/capture/input_generators/attack_point_iterator.py:93]:
```suggestion

                values (List[List[int]]): List of lists of ints that gets
```

**Review by kralka — COMMENTED** [review]
Thank you, PTAL at the suggestions.

File: tests/capture/input_generators/test_attack_point_iterator.py (hunk 1 above)
Comment (LucaTomasko) [reply tests/capture/input_generators/test_attack_point_iterator.py]: Is this not good yet?

Comment (kralka) [reply tests/capture/input_generators/test_attack_point_iterator.py]: Ah I have not resolved yet.

**Review by kralka — APPROVED** [review]
Thank you

**Pull request merged**